### PR TITLE
[react-native] Add Focusable View Prop

### PR DESCRIPTION
**Pull request by NickGerleman**
This was added in 0.62 to the AndroidViewProps Flow type. Add to the TS interface as well. https://github.com/facebook/react-native/blob/0.62-stable/Libraries/Components/View/ViewPropTypes.js

There are some other new props where the right story is less defined.

-----

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -2033,6 +2033,11 @@ export interface ViewPropsAndroid {
      * re-used and re-composited with different parameters. The downside is that this can use up limited video memory, so this prop should be set back to false at the end of the interaction/animation.
      */
     renderToHardwareTextureAndroid?: boolean;
+
+    /**
+     * Whether this `View` should be focusable with a non-touch input device, eg. receive focus with a hardware keyboard.
+     */
```
Comment (ca057) [inline types/react-native/index.d.ts:2039]: Should we add the platform information to the docs?

**Review by ca057 — APPROVED** [review]

**Review by a-tarasyuk — APPROVED** [review]

File: types/react-native/index.d.ts (hunk 1 above)
Comment (alloy) [reply types/react-native/index.d.ts:2039]: These are grouped in a ‘ViewPropsAndroid’ interface to indicate they are Android only. It’s not the greatest, but should be enough to ensure people understand when they can use these props?

**Comment by alloy** [comment]
👍

File: types/react-native/index.d.ts (hunk 1 above)
Comment (ca057) [reply types/react-native/index.d.ts:2039]: Ah, I haven’t seen that, thanks for the explanation.

I think it should be enough when you look into the actual types, but it always pops up in intellisense and isn’t mentioned in the documentation string, so unless you look into the type definition it’s not directly visible. I think having it mentioned in there might increase the visibility. Not sure though if this is a problem at all for other people or if this is part of a larger discussion.

Comment (alloy) [reply types/react-native/index.d.ts:2039]: Until we have a way to only load the typings for the platforms one actually targets, I’m not opposed to over-communicating it, so by all means do add it 👍

**Pull request merged**